### ISSUE TITLE
Wemo Binding causes an IllegalArgumentException on OSRAM Lightify Flex discovery

### ISSUE THREAD

**Issue by timonreinhard** [eclipse/smarthome#3183]
Wemo Binding 0.9.0.b2 (on openHAB 2.0.0.b5) causes an `IllegalArgumentException` when an OSRAM Lightify Flex device is paired with the discovered Wemo Link bridge. As a result, not any of the devices connected to the bridge get found.

```
2017-03-24 18:08:03.923 [ERROR] [o.discovery.WemoLinkDiscoveryService] - Failed to parse endDevices for bridge 'wemo:bridge:Bridge-1_0-231443B000007A'
java.lang.IllegalArgumentException: UID segment 'Flex RGBW' contains invalid characters. Each segment of the UID must match the pattern [A-Za-z0-9_-]*.
	at org.eclipse.smarthome.core.thing.UID.validateSegment(UID.java:91)[103:org.eclipse.smarthome.core.thing:0.9.0.b2]
	at org.eclipse.smarthome.core.thing.UID.<init>(UID.java:69)[103:org.eclipse.smarthome.core.thing:0.9.0.b2]
	at org.eclipse.smarthome.core.thing.ThingTypeUID.<init>(ThingTypeUID.java:31)[103:org.eclipse.smarthome.core.thing:0.9.0.b2]
	at org.eclipse.smarthome.binding.wemo.discovery.WemoLinkDiscoveryService.startScan(WemoLinkDiscoveryService.java:189)[193:org.eclipse.smarthome.binding.wemo:0.9.0.b2]
	at org.eclipse.smarthome.binding.wemo.discovery.WemoLinkDiscoveryService$WemoLinkScan.run(WemoLinkDiscoveryService.java:283)[193:org.eclipse.smarthome.binding.wemo:0.9.0.b2]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)[:1.8.0_65]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)[:1.8.0_65]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)[:1.8.0_65]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)[:1.8.0_65]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)[:1.8.0_65]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)[:1.8.0_65]
	at java.lang.Thread.run(Thread.java:745)[:1.8.0_65]
```

**Comment by hmerk** [comment]
Only original WeMo LED bulbs are supported by the binding.

**Comment by timonreinhard** [comment]
This is what I understood from the docs, right. I have two original Wemo bulbs and the Osram mentioned above in my setup. My expectation was that the Wemo bulbs get discovered while the Osram will be ignored. I'm however facing that no bulb is found and the exception above is thrown every couple of seconds, even after removing the Wemo Link thing.

**Comment by hmerk** [comment]
Ok, this sounds clearly like a bug. I will check this asap.

**Comment by hmerk** [comment]
To track this down to the root caus, could you please set the WeMo binding to log level trace
**log:set TRACE org.eclipse.smarthome.binding.wemo**
and post your discovery output.

**Comment by timonreinhard** [comment]
```
2017-03-26 19:13:19.868 [TRACE] [o.discovery.WemoLinkDiscoveryService] - Starting WeMoEndDevice discovery on WeMo Link wemo:bridge:Bridge-1_0-231443B010007A
2017-03-26 19:13:19.868 [TRACE] [o.discovery.WemoLinkDiscoveryService] - devUDN = 'uuid:Bridge-1_0-231443B010007A'
2017-03-26 19:13:19.933 [TRACE] [o.discovery.WemoLinkDiscoveryService] - endDeviceRequest answered '<s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/" s:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/"><s:Body>
<u:GetEndDevicesResponse xmlns:u="urn:Belkin:service:bridge:1">
<DeviceLists>&lt;?xml version=&quot;1.0&quot; encoding=&quot;utf-8&quot;?&gt;&lt;DeviceLists&gt;&lt;DeviceList&gt;&lt;DeviceListType&gt;Paired&lt;/DeviceListType&gt;&lt;DeviceInfos&gt;&lt;DeviceInfo&gt;&lt;DeviceIndex&gt;0&lt;/DeviceIndex&gt;&lt;DeviceID&gt;8418260000CAB841&lt;/DeviceID&gt;&lt;FriendlyName&gt;TV-Backlight&lt;/FriendlyName&gt;&lt;IconVersion&gt;1&lt;/IconVersion&gt;&lt;FirmwareVersion&gt;03&lt;/FirmwareVersion&gt;&lt;CapabilityIDs&gt;10006,10008,10300,30008,30009,3000A,30301&lt;/CapabilityIDs&gt;&lt;CurrentState&gt;0,255:0,36532:24789:0,,,,370:0&lt;/CurrentState&gt;&lt;Manufacturer&gt;OSRAM&lt;/Manufacturer&gt;&lt;ModelCode&gt;Flex RGBW&lt;/ModelCode&gt;&lt;productName&gt;FlexBulb&lt;/productName&gt;&lt;WeMoCertified&gt;NO&lt;/WeMoCertified&gt;&lt;/DeviceInfo&gt;&lt;DeviceInfo&gt;&lt;DeviceIndex&gt;2&lt;/DeviceIndex&gt;&lt;DeviceID&gt;94103EA2B278263F&lt;/DeviceID&gt;&lt;FriendlyName&gt;Foo&lt;/FriendlyName&gt;&lt;IconVersion&gt;1&lt;/IconVersion&gt;&lt;FirmwareVersion&gt;83&lt;/FirmwareVersion&gt;&lt;CapabilityIDs&gt;10006,10008,30008,30009,3000A&lt;/CapabilityIDs&gt;&lt;CurrentState&gt;0,255:0,,,&lt;/CurrentState&gt;&lt;Manufacturer&gt;MRVL&lt;/Manufacturer&gt;&lt;ModelCode&gt;MZ100&lt;/ModelCode&gt;&lt;productName&gt;Lighting&lt;/productName&gt;&lt;WeMoCertified&gt;YES&lt;/WeMoCertified&gt;&lt;/DeviceInfo&gt;&lt;/DeviceInfos&gt;&lt;GroupInfos&gt;&lt;GroupInfo&gt;&lt;GroupID&gt;1432253402&lt;/GroupID&gt;&lt;GroupName&gt;Bar&lt;/GroupName&gt;&lt;GroupCapabilityIDs&gt;10006,10008,30008,30009,3000A&lt;/GroupCapabilityIDs&gt;&lt;GroupCapabilityValues&gt;1,255:0,0:0,,&lt;/GroupCapabilityValues&gt;&lt;DeviceInfos&gt;&lt;DeviceInfo&gt;&lt;DeviceIndex&gt;0&lt;/DeviceIndex&gt;&lt;DeviceID&gt;94103EA2B27808EC&lt;/DeviceID&gt;&lt;FriendlyName&gt;Lightbulb 01&lt;/FriendlyName&gt;&lt;IconVersion&gt;1&lt;/IconVersion&gt;&lt;FirmwareVersion&gt;83&lt;/FirmwareVersion&gt;&lt;CapabilityIDs&gt;10006,10008,30008,30009,3000A&lt;/CapabilityIDs&gt;&lt;CurrentState&gt;1,255:0,,,&lt;/CurrentState&gt;&lt;Manufacturer&gt;MRVL&lt;/Manufacturer&gt;&lt;ModelCode&gt;MZ100&lt;/ModelCode&gt;&lt;productName&gt;Lighting&lt;/productName&gt;&lt;WeMoCertified&gt;YES&lt;/WeMoCertified&gt;&lt;/DeviceInfo&gt;&lt;DeviceInfo&gt;&lt;DeviceIndex&gt;1&lt;/DeviceIndex&gt;&lt;DeviceID&gt;94103EA2B2787231&lt;/DeviceID&gt;&lt;FriendlyName&gt;Lightbulb 03&lt;/FriendlyName&gt;&lt;IconVersion&gt;1&lt;/IconVersion&gt;&lt;FirmwareVersion&gt;83&lt;/FirmwareVersion&gt;&lt;CapabilityIDs&gt;10006,10008,30008,30009,3000A&lt;/CapabilityIDs&gt;&lt;CurrentState&gt;1,255:0,,,&lt;/CurrentState&gt;&lt;Manufacturer&gt;MRVL&lt;/Manufacturer&gt;&lt;ModelCode&gt;MZ100&lt;/ModelCode&gt;&lt;productName&gt;Lighting&lt;/productName&gt;&lt;WeMoCertified&gt;YES&lt;/WeMoCertified&gt;&lt;/DeviceInfo&gt;&lt;/DeviceInfos&gt;&lt;/GroupInfo&gt;&lt;/GroupInfos&gt;&lt;/DeviceList&gt;&lt;/DeviceLists&gt;
</DeviceLists>
</u:GetEndDevicesResponse>
</s:Body> </s:Envelope>'
2017-03-26 19:13:19.936 [TRACE] [o.discovery.WemoLinkDiscoveryService] - DeviceIndex: 0
2017-03-26 19:13:19.936 [TRACE] [o.discovery.WemoLinkDiscoveryService] - DeviceID: 8418260000CAB841
2017-03-26 19:13:19.936 [TRACE] [o.discovery.WemoLinkDiscoveryService] - FriendlyName: TV-Backlight
2017-03-26 19:13:19.936 [TRACE] [o.discovery.WemoLinkDiscoveryService] - Manufacturer: OSRAM
2017-03-26 19:13:19.939 [TRACE] [o.discovery.WemoLinkDiscoveryService] - ModelCode: Flex RGBW
2017-03-26 19:13:19.939 [ERROR] [o.discovery.WemoLinkDiscoveryService] - Failed to parse endDevices for bridge 'wemo:bridge:Bridge-1_0-231443B010007A'
java.lang.IllegalArgumentException: UID segment 'Flex RGBW' contains invalid characters. Each segment of the UID must match the pattern [A-Za-z0-9_-]*.
	at org.eclipse.smarthome.core.thing.UID.validateSegment(UID.java:91)[103:org.eclipse.smarthome.core.thing:0.9.0.201611161058]
	at org.eclipse.smarthome.core.thing.UID.<init>(UID.java:69)[103:org.eclipse.smarthome.core.thing:0.9.0.201611161058]
	at org.eclipse.smarthome.core.thing.ThingTypeUID.<init>(ThingTypeUID.java:31)[103:org.eclipse.smarthome.core.thing:0.9.0.201611161058]
	at org.eclipse.smarthome.binding.wemo.discovery.WemoLinkDiscoveryService.startScan(WemoLinkDiscoveryService.java:189)[196:org.eclipse.smarthome.binding.wemo:0.9.0.201611161058]
	at org.eclipse.smarthome.binding.wemo.discovery.WemoLinkDiscoveryService$WemoLinkScan.run(WemoLinkDiscoveryService.java:283)[196:org.eclipse.smarthome.binding.wemo:0.9.0.201611161058]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)[:1.8.0_112]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)[:1.8.0_112]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)[:1.8.0_112]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)[:1.8.0_112]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)[:1.8.0_112]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)[:1.8.0_112]
	at java.lang.Thread.run(Thread.java:745)[:1.8.0_112]
```

**Comment by hmerk** [comment]
Should be fixed in #3188

**Comment by maggu2810** [comment]
Should be closed by #3188.
@hmerk Can you have a look at https://help.github.com/articles/closing-issues-via-commit-messages/ :wink:

**Comment by hmerk** [comment]
@maggu2810 of course, I will check this.
